src/js/gutenberg-plugins/content-resizing-plugin.js

```diff
@@ -0,0 +1,569 @@
+				const __textArray = await getResizedContent();
+				setTextArray( __textArray );
+				setIsModalOpen( true );
+			} )();
+		}
+	}, [ isResizing, getResizedContent ] );
+
+	// Resets to default states.
+	function resetStates() {
+		setSelectedBlock( null );
+		setTextArray( [] );
+		setIsModalOpen( false );
+		dispatch( resizeContentStore ).setResizingType( null );
+		isProcessAnimating = false;
+	}
+
+	/**
+	 * Triggered when either `Grow content` or `Shrink content` is clicked from
+	 * the Block's "more options" menu.
+	 *
+	 * @return {void}
+	 */
+	async function resizeContent() {
+		const block = select( blockEditorStore ).getSelectedBlock();
+		const blockContent = block.attributes.content ?? '';
+
+		setSelectedBlock( block );
+		setBlockContentAsPlainText( toPlainText( blockContent ) );
+		setIsResizing( true );
+	}
+
+	/**
+	 * The AJAX callback that returns with an array of suggestion.
+	 *
+	 * @return {Array} Array of suggestions.
+	 */
+	async function getResizedContent() {
+		let __textArray = [];
+		const apiUrl = `${ wpApiSettings.root }classifai/v1/openai/resize-content`;
+		const postId = select( editorStore ).getCurrentPostId();
+		const formData = new FormData();
+
+		formData.append( 'id', postId );
+		formData.append( 'content', blockContentAsPlainText );
+		formData.append( 'resize_type', resizingType );
+
+		dispatch( resizeContentStore ).setClientId( selectedBlock.clientId );
+
+		const response = await fetch( apiUrl, {
+			method: 'POST',
+			body: formData,
+			headers: new Headers( {
+				'X-WP-Nonce': wpApiSettings.nonce,
+			} ),
+		} );
+
+		if ( 200 === response.status ) {
+			__textArray = await response.json();
+		} else {
+			setIsResizing( false );
+			dispatch( resizeContentStore ).setClientId( '' );
+			resetStates();
+		}
+
+		setIsResizing( false );
+		dispatch( resizeContentStore ).setClientId( '' );
+
+		return __textArray;
+	}
+
+	/**
+	 * Updates the text selection.
+	 *
+	 * @param {string} updateWith The content that will be used to replace the selection.
+	 */
+	function updateContent( updateWith ) {
+		dispatch( blockEditorStore ).updateBlockAttributes(
+			selectedBlock.clientId,
+			{
+				content: updateWith,
+			}
+		);
+
+		dispatch( blockEditorStore ).selectionChange(
+			selectedBlock.clientId,
+			'content',
+			0,
+			updateWith.length
+		);
+		resetStates();
+	}
+
+	// We don't want to use the reszing feature when multiple blocks are selected.
+	// Nor do we want to use it when a block's content resizing is under process.
+	if ( isMultiBlocksSelected || isResizing ) {
+		return null;
+	}
+
+	// Result Modal JSX.
+	const suggestionModal = ! isResizing && textArray.length && isModalOpen && (
+		<Modal
+			title={ __( 'Suggestions', 'classifai' ) }
+			isFullScreen={ false }
+			className="classifai-content-resize__suggestion-modal"
+			onRequestClose={ () => {
+				setIsModalOpen( false );
+				resetStates();
+			} }
+		>
+			<p>{ __( 'Click on a row to apply:' ) }</p>
```

Comment:
Not sure we need this paragraph. It's not super accurate (you're not clicking on the row but on the button) and I think it's clear enough what you need to do

Reply:
So looked at the failing E2E test and the issue was we were clicking on the row, not the button. I've fixed that in [141ea55](https://github.com/10up/classifai/pull/532/commits/141ea558c9ea8c64d0853f58c2effb94f2204fe4) but I guess wondering now if the intention is to allow the user to click on the entire row to insert? Or is clicking on the button good enough?

Reply:
@Sidsector9 Just flagging this again. Let me know if this is a feature that we want and just isn't working. Otherwise I'd suggest we remove this text